### PR TITLE
fix(relay): Fix internal ip check for is_internal_relay

### PR DESCRIPTION
**Pull request by BYK**
`settings.INTERNAL_IPS` is meant to be a list of IP networks, not individual IPs, hence the `is_internal_relay` check was not working (see getsentry/onpremise#572). This PR fixes the issue by replacing the manual and incorrect check with `sentry.auth.system.is_internal_ip`. This also means we are now checking against `INTERNAL_SYSTEM_IPS` instead of `INTERNAL_IPS` which is more accurate.

### CODE REVIEW

**Review by untitaker — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]
Good catch, thanks!

File: src/sentry/api/endpoints/relay_register.py [hunk 1]
```diff
@@ -42,13 +43,10 @@ def is_internal_relay(request, public_key):
...
-    if (
-        settings.DEBUG
-        or request.META.get("REMOTE_ADDR", None) in settings.INTERNAL_IPS
-        or public_key in settings.SENTRY_RELAY_WHITELIST_PK
-    ):
+    if settings.DEBUG or public_key in settings.SENTRY_RELAY_WHITELIST_PK:
         return True
-    return False
+
+    return is_internal_ip(is_internal_ip)
```
Comment (jan-auer) [inline src/sentry/api/endpoints/relay_register.py:49]: `is_internal_ip` uses [`INTERNAL_SYSTEM_IPS`](https://github.com/getsentry/sentry/blob/bfe27f43dc28d308546ad03446e24c1ebf9b6c9a/src/sentry/conf/server.py#L83-L99). Is that what you want?

You might need to add the docker network to that list in the onpremise config.

Comment (BYK) [reply src/sentry/api/endpoints/relay_register.py:49]: Already done: [`9885965` (#572)](https://github.com/getsentry/onpremise/pull/572/commits/988596581bac49f9cbab283a2fa5b1e8c1de7ade)

**Review by jan-auer — APPROVED** [review]
LGTM.

**Pull request merged**